### ISSUE TITLE
Subscription-based non-web application gracefully shuts down without waiting for messages

### ISSUE THREAD

**Issue by elefeint** [spring-cloud/spring-cloud-gcp#2014]
The yak shaving, it continueth.

After #2010 went in, the Pub/Sub starter thread pools get shut down a little too cleanly. I've planned to document (#2012) the behavior for non-web apps, but it so happens that Spring Cloud Stream's sinks are often non-web apps. So, if you start the sink app from our samples now, it starts up, looks around, does not find anything interesting, and shuts down.

@artembilan What is expected behavior of a Spring Integration application without any web components that listens to a queue? Intuitively, it seems that such an application should stay up. 

What if we provide a daemon thread pool for just `PubSubTemplate` but provide a non-daemon thread pool whenever Spring Integration is detected together with Pub/Sub?

**Comment by elefeint** [comment]
I thought GitHub outage ate this issue forever. Closing it, as it's a duplicate of #2015 that has the actual discussion.
